ClientFactory/PluginClientFactory.php

```diff
@@ -16,11 +16,12 @@
      * @param Plugin[]      $plugins
      * @param ClientFactory $factory
      * @param array         $config
+     * @param array         $pluginClientOptions
```

Comment:
can you add a short text that explains the $config and $pluginClientOptions? as its two sets of options, it would help to know what each is to keep them cleanly separated. saying that $pluginClientOptions is forwareded to PluginClient is good enough for me.